src/App.js

```diff
@@ -1,26 +1,106 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { Component } from 'react';
+import Header from './Header.js'
+import Footer from './Footer.js';
+// import ImageList from './ImageList';
+import images from './data.js';
 import './App.css';
+import ImageItem from './ImageItem.js';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+export default class App extends Component {
+  // init state as null -- nothing selected
+  state = { 
+    // hasBeenChanged: false,
+    selectedKeyword: null,
+    // selectedHorns: null
+  }
+
+  // create function to update state based on change to `select`
+
+  handleChangedKeyword = (e) => {
+    this.setState({ 
+      // hasBeenChanged: true,
+      selectedKeyword: e.target.value,
+      // selectedHorns: null 
+    });
+  }
+
+  // handleChangedHorns = (e) => {
+  //   const parsedValue = parseInt(e.target.value);
+  //   this.setState({ 
+  //     hasBeenChanged: true,
+  //     selectedKeyword: null, 
+  //     selectedHorns: parsedValue 
+  //   });
+  // }
+
+  render() {
```

Comment:
Here is the solution I found for not showing duplicates in your drop-down:

```js
render() {
    images.forEach(image => {
      if(this.state.options.includes(image.keyword)) return;

      this.setState({options: [...this.state.options, image.keyword]})
    })

    return (
      <>
        <Header></Header>
        <main>
          <section>
          <p>Filter by keyword:</p>
          <select onChange={ this.handleChangedKeyword }>

              {this.state.options.map(option => (<option value={option}>{option}</option>))}
```

Reply:
Thanks, Jacob!